depthai_ros_driver/src/conversion.cpp

```diff
@@ -0,0 +1,75 @@
+#include "depthai_ros_driver/conversion.hpp"
+#include <opencv2/imgproc.hpp>
+
+cv::Mat rr::convertImg(const depthai_datatype_msgs::RawImgFrame& input) {
+    // Note: stolen from depthai-core/src/opencv/ImgFrame.cpp
+    cv::Mat mat;
+    cv::Size size = {0, 0};
+    int type = 0;
+
+    switch (static_cast<dai::RawImgFrame::Type>(input.fb.type)) {
+        case dai::RawImgFrame::Type::RGB888i:
+        case dai::RawImgFrame::Type::BGR888i:
+        case dai::RawImgFrame::Type::BGR888p:
+        case dai::RawImgFrame::Type::RGB888p:
+            size = cv::Size(input.fb.width, input.fb.height);
+            type = CV_8UC3;
+            break;
+
+        case dai::RawImgFrame::Type::YUV420p:
+        case dai::RawImgFrame::Type::NV12:
+        case dai::RawImgFrame::Type::NV21:
+            size = cv::Size(input.fb.width, input.fb.height * 3 / 2);
+            type = CV_8UC1;
+            break;
+
+        case dai::RawImgFrame::Type::RAW8:
+        case dai::RawImgFrame::Type::GRAY8:
+            size = cv::Size(input.fb.width, input.fb.height);
+            type = CV_8UC1;
+            break;
+
+        case dai::RawImgFrame::Type::GRAYF16:
+            size = cv::Size(input.fb.width, input.fb.height);
+            type = CV_16FC1;
+            break;
+
+        case dai::RawImgFrame::Type::RAW16:
+            size = cv::Size(input.fb.width, input.fb.height);
+            type = CV_16UC1;
+            break;
+
+        case dai::RawImgFrame::Type::RGBF16F16F16i:
+        case dai::RawImgFrame::Type::BGRF16F16F16i:
+        case dai::RawImgFrame::Type::RGBF16F16F16p:
+        case dai::RawImgFrame::Type::BGRF16F16F16p:
+            size = cv::Size(input.fb.width, input.fb.height);
+            type = CV_16FC3;
+            break;
+
+        case dai::RawImgFrame::Type::BITSTREAM:
+        default:
+            size = cv::Size(static_cast<int>(input.data.size()), 1);
+            type = CV_8UC1;
+            break;
+    }
+
+    // Check if enough data
+    long requiredSize = CV_ELEM_SIZE(type) * size.area();
+    long actualSize = static_cast<long>(input.data.size());
+    if (actualSize < requiredSize) {
```

Comment:
What if actualSize > requiredSize?

Reply:
Stolen from here: https://github.com/luxonis/depthai-core/blob/744b32678a0c418dd78cd47736e1eb50a86f9fab/src/opencv/ImgFrame.cpp#L73-L76

Their code is empty, so I ignore the actualSize > requiredSize?

Reply:
Nah. Let's throw an error in either case 😆 and see what happens